### PR TITLE
Update the Hashable implementations.

### PR DESCRIPTION
**Pull request by andersio**
In one scenario, synthesised Hashable conformance can be used. In another scenario, the manual computation can be replaced by combining values via the `Hasher`.

### CODE REVIEW

**Review by inamiy — APPROVED** [review]

**Comment by RuiAAPeres** [comment]
Why is that the case? Both `MovePath` and `Changeset.Move` seem to have a `destination` and a `source`

**Comment by andersio** [comment]
@RuiAAPeres `Changeset.Move` has also a flag `isMutated`. While I am fairly certain that it is not necessary to exclude `isMutated` from hashing, I'd prefer to preserve the current behavior for a quick Swift 5.0 release, and perhaps change that when we revisit the API ergonomics based on experience with Bento. :)

**Comment by RuiAAPeres** [comment]
Sorry for my ignorance, but I still don't understand your answer. As in, why can't we use the synthesized version for`MovePath` or use the custom implementation for `Changeset.Move `? Also, why do we need to rush a Swift 5.0 release?

**Comment by andersio** [comment]
`Changeset.Move` is clearly hashing all its properties, so that can be replaced by the synthesised version. `MovePath` I am not quite sure, and needs to review it.
